File: projects/backend/src/sun.rs
use crate::Location;

pub fn sun_position(date_time: chrono::DateTime<chrono::Utc>, location: Location) -> SunPosition {
    let pos = sun::pos(date_time.timestamp_millis(), location.lat, location.lon);
    return SunPosition { altitude: pos.altitude.to_degrees(), azimuth: pos.azimuth.to_degrees() };
}


#[derive(Debug)]
pub struct SunPosition {
    pub altitude: f64,
    pub azimuth: f64,
}

File: projects/backend/src/backend/sun.clj
(ns backend.sun
  (:import (org.shredzone.commons.suncalc SunPosition)))

(defn position [{:keys [datetime]
                 {:keys [lat lon]} :location}]
  (let [pos (-> (SunPosition/compute)
                (.on datetime)
                (.at lat lon)
                (.execute))]
    {:azimuth (.getAzimuth pos)
     :altitude (.getAltitude pos)}))
